src/components/ProviderList/index.js

```diff
@@ -25,13 +25,16 @@ function ProviderList() {
   };
 
   const resultsLabel = () => {
-    switch (mode) {
-      case "geo":
-      case "postcode":
-        return "closest to you";
-      default:
-        return "across the country";
+    if (selectedPostcode !== null || selectedLocation !== "All") {
+      return `closest to ${selectedPostcode || selectedLocation}`
     }
+    if (selectedLocation === "All") {
+      return `across the country`
+    }
+    // never reaches this case, as selectedLocation must either be All OR a specific location.
+    // if (mode === "geo") {
+    //   return `closest to you`
+    // }
   };
```

Comment:
I think I'd have kept the `switch` statement; choosing between a set of values is exactly what switches are for.

Would this work?
```
switch (mode) {
      case "geo":
         return "closest to you";
      case "postcode":	
         return `closest to ${selectedPostcode}`
      default:	
        return selectedLocation === "All" ? "across the country" : `closest to ${selectedLocation}`;	
    }

Reply:
also - might be beneficial to keep `selectedPostcode` in the GeoProvider, and set it when setting the postcode. That way, we know it'll be defined when `mode === 'postcode'`

Reply:
That does work in theory, although as far as I can tell the geo case always passes here.

Having geo as the first case means if geo is active you'll lose all of the context surrounding your results when you search all providers, search location and enter a postcode. In all of these cases now `closest to you` is given as the results info text.

It was because of this I thought about dropping the switch entirely - with this approach I managed to get 3 tailored results info, as opposed to a blanket `closest to you` that I currently get when implementing this suggested switch.

Reply:
I see what you mean. There's been a longstanding issue that the string would show 'closest to you' when it shouldn't. This is an issue in the geo provider, geo mode being set before it's supposed to. 

My issue is that right now on the deploy preview, it's getting the results near me, but then saying 'showing 10 venues across the country'. Which is completely wrong (almost the opposite of the original issue which is it says 'showing 700 venues closest to you'). I think this is worse (because it makes it look like we might only have 10 results).